tools/clone-tutorial.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python3
+
+
+#  Copyright (c) 2023 Confluent
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import argparse
+import glob
+import os
+import rtyaml
+import shutil
+
+tools_home = os.path.realpath(os.path.dirname(__file__))
+kt_home = os.path.dirname(tools_home)
+print("Using [{dir}] as the base directory for the tutorial".format(dir=kt_home))
+
+tutorial_types = ['ksql', 'kstreams', 'kafka', 'confluent', 'flinksql']
+
+parser = argparse.ArgumentParser(description='Generates a clone of an existing tutorial updated for the new name')
+parser.add_argument('--orig-name', required=True,
+                    help='The name of the tutorial to clone, if a subtype i.e. ksqlDB the provide path name '
+                         'session-windows/ksql')
+parser.add_argument('--new-name', required=True,
+                    help='The name of the new tutorial if a subtype i.e flink-sql provide path name '
+                         'session-windows/flinksql')
+parser.add_argument('--orig-main-class', help='Name of original main class (applicable for Java/Flink)')
+parser.add_argument('--new-main-class', help='Name of new main class (applicable for Java/Flink)')
+parser.add_argument('--semaphore-test-name', required=True, help='The name of the test for semaphore.yml file')
+parser.add_argument('--permalink', help='The permalink for the tutorial - more descriptive link')
+# parser.add_argument('--json-file', help="The path to a json file containing these configurations")
+parser.add_argument('--debug', default=False, help='The permalink for the tutorial')
```

Comment:
optional: use `logging` module rather than `print` and set the level via argument (e.g. see [this](https://stackoverflow.com/a/20663028/5587460) SO)

the primary benefit would then be that rather than having to check and `print` like:
```
if args.debug:
    print("...")
```
it'd be:
```
logging.debug("...")
```
Not a huge benefit today but would be nice to have for maintenance and evolving this

Reply:
yep went with logging - much better than print statements